server/etcdserver/api/v3discovery/discovery_test.go

```diff
@@ -0,0 +1,651 @@
+package v3discovery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	"go.etcd.io/etcd/client/pkg/v3/types"
+	"go.etcd.io/etcd/client/v3"
+
+	"github.com/jonboulle/clockwork"
+	"go.uber.org/zap"
+)
+
+var (
+	clusterFakeToken = "fakeToken"
+
+	actualMemberInfo = []memberInfo{
```

Comment:
I understand that it is nice to deduplicate code, in this case test cases. However I would that for tests repetition is a benefit because it makes each test independent. Imagine that one of the cases fails in single test. Should you fix the test or the input data? 

Other downside of this is fact that distance between test code and input data, Looking at the test code it's hard to reason if the test should pass because the input for the test is on the other side of file. Please consider moving member info to each test. If you are worried about code duplication please check out Factory pattern.

Reply:
Thanks for the comment and I tend to agree with you. Although the data is identical for multiple test cases for now, they should be maintained & updated separately. 

I thought about factory pattern + abstract pattern. In the first step, the factory creates exactly the same data for all test cases; afterwards, the modify / refine the data using the abstract class/utility. It can reduce the lines of code, but it introduces some complexity. So I don't like it for now, because the test cases are simple enough at the moment. 

Just moved the data into each test case.  Thanks again.